tfrecord/iterator_utils.py

```diff
@@ -27,19 +28,35 @@ def sample_iterators(iterators: typing.List[typing.Iterator],
 
     ratios: list of int
         The ratios with which to sample each iterator.
+    
+    infinite: bool, optional, default=True
+        Whether the returned iterator should be infinite or not
 
     Yields:
     -------
     item: Any
         Decoded bytes of features into its respective data types from
         an iterator (based off their sampling ratio).
     """
-    iterators = [cycle(iterator) for iterator in iterators]
+    if infinite:
+        iterators = [cycle(iterator) for iterator in iterators]
+    else:
+        iterators = [iterator() for iterator in iterators]
     ratios = np.array(ratios)
     ratios = ratios / ratios.sum()
-    while True:
+    while iterators:
         choice = np.random.choice(len(ratios), p=ratios)
-        yield next(iterators[choice])
+        if infinite:
+            yield next(iterators[choice])
```

Comment:
The if condition is not required, you can just keep the else part.

Reply:
Yes, you're right, I added it there since there is a penalty hit for entering `try..except` block, but I guess the difference is negligible enough to not really care.